scripts/import-linter.ts

```diff
@@ -0,0 +1,145 @@
+/// <reference lib="esnext" />
+
+const path: typeof import('path') = require('path')
+const ts: typeof import('ts-morph') = require('ts-morph')
+import { SourceFile, Project } from 'ts-morph'
+
+const project = new ts.Project({
+    addFilesFromTsConfig: true,
+    tsConfigFilePath: path.join(__dirname, '../tsconfig.json'),
+})
+
+// for (const each of project.getSourceFiles()) {
+//     console.log(each.getFilePath())
+// }
+
+// Task 1: Ban imports from background page
+const background = project.getSourceFileOrThrow(path.join(__dirname, '../src/background-service.ts'))
+
+const checkedList = new Set<SourceFile>()
+checkedList.add(project.getSourceFileOrThrow(path.join(__dirname, '../src/utils/safeRequire.ts')))
+const diags = checkReferenceRecursive(project, background, getChecker(), checkedList, [background.getFilePath()])
+if (diags) {
+    throw 'There is diagnostics generated by the linter. Please fix them before continue.'
+}
+
+function getChecker(): (path: string) => string {
+    return x => {
+        if (x.includes('@material-ui')) {
+            return `@material-ui appears in the import chain!
+To write a cross context file, use safeMUI() inside the function to import them conditionally.
+`
+        } else if (x === 'react') {
+            return `React appears in the import chain!
+To write a cross context file, use safeReact() inside the function to import them conditionally.
+`
+        }
+        return null
```

Comment:
`null` is not asignable to string type.

Reply:
yeah you are right that file is in the scripts/ folder so the strict null check is not opened